### PR TITLE
migrate to aes in keystore

### PR DESCRIPTION
**Pull request by bobzilladev**
Handling https://github.com/wiglenet/wigle-wifi-wardriving/issues/155 as well as lineage issue, hopefully.

### CODE REVIEW

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/TokenAccess.java [hunk 1]
```diff
@@ -86,6 +92,59 @@ public static void clearApiToken(SharedPreferences prefs) {
...
+            throws KeyStoreException, IOException, NoSuchAlgorithmException, CertificateException,
+            NoSuchPaddingException, InvalidKeyException, UnrecoverableEntryException, IllegalBlockSizeException,
+            BadPaddingException {
+
+        final KeyStore keyStore = getKeyStore();
+        final SecretKey key = (SecretKey) keyStore.getKey(KEYSTORE_WIGLE_CREDS_KEY_V2, null);
+        Cipher encrypt = Cipher.getInstance(AES_CIPHER);
+        encrypt.init(Cipher.ENCRYPT_MODE, key);
+
+        final byte[] input = apiToken.getBytes();
```
Comment (rksh) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/TokenAccess.java]: does apiToken need a null check someplace before this?

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/TokenAccess.java [hunk 2]
```diff
@@ -86,6 +92,59 @@ public static void clearApiToken(SharedPreferences prefs) {
...
+        MainActivity.info("[TOKEN] setApiTokenVersion2 set token length: " + apiToken.length());
+        return true;
+    }
+
+    private static String getApiTokenVersion2(SharedPreferences prefs) throws KeyStoreException, IOException,
+            NoSuchAlgorithmException, CertificateException, NoSuchPaddingException, InvalidKeyException,
+            UnrecoverableEntryException, IllegalBlockSizeException, BadPaddingException,
+            InvalidAlgorithmParameterException {
+
+        if (android.os.Build.VERSION.SDK_INT < Build.VERSION_CODES.M) return null;
```
Comment (rksh) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/TokenAccess.java]: might be worth logging here, JiC

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/TokenAccess.java [hunk 3]
```diff
@@ -53,11 +59,11 @@ public static boolean hasApiToken(SharedPreferences prefs) {
...
                 try {
-                    KeyStore keyStore = KeyStore.getInstance(ANDROID_KEYSTORE);
-                    keyStore.load(null);
+                    final KeyStore keyStore = getKeyStore();
 
                     if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.M) {
-                        if (keyStore.containsAlias(KEYSTORE_WIGLE_CREDS_KEY_V1)) {
+                        if (keyStore.containsAlias(KEYSTORE_WIGLE_CREDS_KEY_V1)
+                                || keyStore.containsAlias(KEYSTORE_WIGLE_CREDS_KEY_V2)) {
                             //TODO: it would be best to test decrypt here, but makes this heavier
```
Comment (bobzilladev) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/TokenAccess.java]: That did occur to me, bias towards Things Working. I didn't want to bite off too much, I whittled down my changes to something reviewable, and try to keep existing paths not modified too much, cuz there's already a lot of branching and I got scared.

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/TokenAccess.java (hunk 1 above)
Comment (bobzilladev) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/TokenAccess.java]: looks like yes, I can add when i get home.

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/TokenAccess.java (hunk 2 above)
Comment (bobzilladev) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/TokenAccess.java]: agreed

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/TokenAccess.java (hunk 3 above)
Comment (rksh) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/TokenAccess.java]: good point. let's test this, maybe do some cleanup/insurance after.

**Comment by bobzilladev** [comment]
pushed commit for comments.

**Comment by rksh** [comment]
ship it.

**Pull request merged**